enhancements/insights/config-api.md

```diff
@@ -0,0 +1,250 @@
+
+```go
+type InsightsSpec struct {
+	// Gathering spec attribute includes all the configuration options related to
+	// gathering of the Insights archive and its uploading to the ingress.
+	GatheringConfig `json:"gatheringConfig,omitempty"`
+}
+
+type InsightsStatus struct {
+	// GatheringStatus provides basic information about the last Insights gathering.
+	GatheringStatus *GatheringStatus `json:"gatheringStatus,omitempty"`
+	// ReportStatus provides general Insights analysis results.
+	ReportStatus *ReportStatus `json:"reportStatus,omitempty"`
+}
+
+type GatheringConfig struct {
+	// EnableObfuscation is flag enabling the global obfuscation of the IP addresses and base domain
+	// in the Insights archive data.
+	EnableObfuscation bool `json:"enableObfuscation,omitempty"`
+	// Enables user to force Insights data gathering by setting a new reason. 
+	// When there is some gathering in the progress then it is interrupted.
+	// When all the gatherers are deactivated by the `DisabledGatherers`, nothing happens.
+	ForceGatherReason string `json:"forceGatherReason,omitempty"`
+	// List of gatherers to be excluded from the gathering. All the gatherers can disabled by providing "all" value.
+	// If all the gatherers are disabled, the Insights operator does not gather any data.
+	DisabledGatherers []string `json:"disabledGatherers,omitempty"`
+}
+
+type GatheringStatus struct {
+	// LastGatheringTime is the last time when Insights gathering finished.
+	LastGatheringTime time.Time `json:"lastGatheringTime,omitempty"`
+	// Reason provides last known reason of gathering. This is helpful
+	// especially when gathering was forced by user
+	Reason string `json:"reason,omitempty"`
+	// Status is waiting or running. If it is running then it means there is some gathering in progress.
+	// If it is waiting then a new gathering can be triggered immediately.
+	Status string `json:"status,omitempty"`
+	// List of active gatherers (and their statuses) in the last gathering.
+	Gatherers []GathererStatus `json:"gatherers,omitempty"`
+}
+
+type ReportStatus struct {
+	// Number of active Insights healthchecks with low severity
+	LowCount int `json:"low,omitempty"`
+	// Number of active Insights healthchecks with moderate severity
+	ModerateCount int `json:"moderate,omitempty"`
+	// Number of active Insights healthchecks with important severity
+	ImportantCount int `json:"important,omitempty"`
+	// Number of active Insights healthchecks with critical severity
+	CriticalCount int `json:"critical,omitempty"`
+	// TotalCount is the count of all active Insights healthchecks
+	TotalCount int `json:"total,omitempty"`
+}
+
+type GathererStatus struct {
+	// Name is the name of the gatherer.
+	Name string `json:"name"`
+	// Errors records any failures while gathering.
+	Errors []string `json:"errors"`
```

Comment:
still resisting to use Conditions over iterating error strings? 

see: https://maelvls.dev/kubernetes-conditions/

Reply:
OK so I tried to imagine how it would look like and it would probably be something like:
```yaml
status:
    gatheringStatus:
      lastGatheringTime: "some timestamp here"
      reason: "common gathering execution"
      status: "waiting"
      gathererStatuses:
        - name: "clusterconfig/nodes"
          conditions:
          - type: "Successful"
            message: "Gathered all the node definitions"
            status: "True"
          recordsCount: 3
          durationMillisecond: 100
        - name: "clusterconfig/machine_sets"
          conditions:
          - type: "Disabled"
            status: "True"
          recordsCount: 0
          durationMillisecond: 0
          ...
```
Currently we have 57 gatherers (and it looks like at least 6 rows per each)  so it will be big wall of status text. Is that OK?

Reply:
Updated. Please take a look @mfojtik. I am thinking that the `GathererCondition.Message` field would be better as a slice of strings......e.g we can multiple errors from one gatherer.